### ISSUE TITLE
Printing/Exporting "note text" to pdf includes the background color of codeBlock

### ISSUE THREAD

**Issue by Waqar144** [pbek/QOwnNotes#1524]
#### Expected behaviour

Printing 'note text' to pdf doesn't include the background color of code-block.

#### Actual behaviour

Printing 'note text' to pdf includes the background color of code-block. Hence looking ugly:

![image](https://user-images.githubusercontent.com/7696024/71569363-57b1a900-2af0-11ea-9fc8-c5cc5573eef7.png)

**Comment by pbek** [comment]
Can't decide if that is a missing feature or a regression. 😆

**Comment by pbek** [comment]
@Waqar144, do you want to fix that?

**Comment by Waqar144** [comment]
Maybe we should just turn off the background somehow when exporting to PDF

**Comment by Waqar144** [comment]
I can try fixing. 😅 But it seems like this one likes you more😂🤣

**Comment by pbek** [comment]
There is the preview PDF export, the preview printing, then export and printing of the selected text, then the html export and that all has to work with dark mode and non-dark mode 😅

**Comment by Waqar144** [comment]
If you draw the rectangle with a separate element like I did the first time. This probably won't happen 😜😂

**Comment by pbek** [comment]
I just realised we weren't talking about the preview export, but the note text export. :grimacing:

**Comment by pbek** [comment]
For some reason I don't have colors in my PDF...

![image](https://user-images.githubusercontent.com/1798101/71589874-c7db2180-2b26-11ea-83e5-87b08e1ca25d.png)

**Comment by Waqar144** [comment]
> I just realised we weren't talking about the preview export, but the note text export. 😬

The fix should be simple,... I think. There was a pretty similar issue recently

**Comment by Waqar144** [comment]
> For some reason I don't have colors in my PDF...
> 
> ![image](https://user-images.githubusercontent.com/1798101/71589874-c7db2180-2b26-11ea-83e5-87b08e1ca25d.png)

Did you check your printer settings? Are you on Linux right now?

**Comment by pbek** [comment]
> Did you check your printer settings? Are you on Linux right now?

Yes, Linux. Maybe because I've a black/white laser printer installed on that computer.

**Comment by Waqar144** [comment]
> > Did you check your printer settings? Are you on Linux right now?
> 
> Yes, Linux. Maybe because I've a black/white laser printer installed on that computer.

Did you try exporting to PDF as well?

**Comment by pbek** [comment]
> Did you try exporting to PDF as well?

Yes, that was a pdf export.

**Comment by pbek** [comment]
I don't know if there is an easy way to get around the background color. We basically just are "printing" the content of the `QTextDocument` in `MainWindow::on_action_Export_note_as_PDF_text_triggered()`

**Comment by pbek** [comment]
> If you draw the rectangle with a separate element like I did the first time. This probably won't happen

Are you talking about the additional style for the background color? Beside being confusing I don't think that would any difference if a schema also uses a "normal" background color.

**Comment by Waqar144** [comment]
> I don't know if there is an easy way to get around the background color. We basically just are "printing" the content of the `QTextDocument` in `MainWindow::on_action_Export_note_as_PDF_text_triggered()`.

There is a way. Just set background color to false but let it have a value(in schema). It doesn't need to be set to true to work.

**Comment by pbek** [comment]
> There is a way. Just set background color to false but let it have a value(in schema). It doesn't need to be set to true to work. I think this should work. What do you think?

But then there would be almost no contrast if there are light foreground colors used.

**Comment by Waqar144** [comment]
@pbek We will still have the rectangle as background. Visually it will be the same as it is right now(in the editor).

**Comment by pbek** [comment]
The rectangles don't seem to be in the PDF.

**Comment by Waqar144** [comment]
> The rectangles don't seem to be in the PDF.

Yes, they won't be there since that part is just a visual effect on the editor.

Though you can get really nice rectangles in the preview export 😅😅

On a side note, we can have a custom processor which filters a document and does all the necessary styling and adding rectangles and stuff before exporting.

**Comment by pbek** [comment]
## 26.5.6

- Fixed printing and exporting the **note text** to PDF while using a dark editor
  color schema by rendering the exported text with the built-in Light editor
  schema instead (for [#1524](https://github.com/pbek/QOwnNotes/issues/1524))
